src/org/jitsi/impl/neomedia/recording/SynchronizerImpl.java

```diff
@@ -324,21 +321,16 @@ private SSRCDesc getSSRCDesc(long ssrc)
      */
     private Endpoint getEndpoint(String endpointId)
     {
-        Endpoint endpoint = endpoints.get(endpointId);
-        if (endpoint == null)
+        synchronized (endpoints)
         {
-            synchronized (endpoints)
+            Endpoint endpoint = endpoints.get(endpointId);
```

Comment:
Unlike ssrcs, there's no protected block that does only reading so I don't think that you want the reads protected. Anyway, I'd like to mention getLocalTime again because there's an unprotected read from endpoints. Some analysis tools would flag such a synchronization as inconsistent.